### PR TITLE
CB-11661: (ios) Add mandatory iOS 10 privacy description

### PR DESCRIPTION
**Pull request by mhartington**
### Platforms affected
iOS

### What does this PR do?
Adds the mandatory NSCameraUsageDescription key to the info.plist, allowing to pass the value as a variable.
Documented the usage of the variable on plugin install as iOS quirk

### Checklist
- [x] [ICLA](http://www.apache.org/licenses/icla.txt) has been signed and submitted to secretary@apache.org.
- [x] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [x] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [ ] Added automated test coverage as appropriate for this change.

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -150,6 +150,11 @@
          <framework src="CoreGraphics.framework" />
          <framework src="AVFoundation.framework" />
 
+         <preference name="CAMERA_USAGE_DESCRIPTION" default=" " />
+         <config-file target="*-Info.plist" parent="NSCameraUsageDescription">
+             <string>$CAMERA_USAGE_DESCRIPTION</string>
+         </config-file>
+
          <config-file target="*-Info.plist" parent="NSLocationWhenInUseUsageDescription">
```
Comment (pke) [inline plugin.xml]: What has this to do in the Camera Plugin?

**Comment by pke** [comment]
Is there a way to localise such texts in `plist`?

File: plugin.xml (hunk 1 above)
Comment (vladimir-kotikov) [inline plugin.xml]: @pke, I believe this is for `CameraUsesGeolocation` [preference](https://github.com/apache/cordova-plugin-camera#preferences-ios)

**Comment by cordova-qa** [comment]
Cordova CI Build has one or more failures. 

**Commit**     - [Link](https://github.com/apache/cordova-plugin-camera/pull/229/commits/8ead470d752fdc50f3d315fb3aeb3fa709ce3f45)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 8.1 Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-8.1-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-8.1-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-8.1-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-8.1-store/artifact/) |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-10-store/artifact/) |
| [Windows 8.1 Phone]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-8.1-phone/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-8.1-phone/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-8.1-phone/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=windows-8.1-phone/artifact/) |
| [iOS]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=ios/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=ios/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=ios/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=ios/artifact/) |
| [Android]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=android/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=android/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=android/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-camera-pr/60//PLATFORM=android/artifact/) |

**Comment by stevengill** [comment]
Merged

**Comment by possan** [comment]
This works great, but it seems you also need to add the plist item `NSPhotoLibraryUsageDescription` for accessing the photo library...

**Comment by jacquesdev** [comment]
@possan - I will try and a PR for this soon because it's currently crashing my app at the moment, if I get it to work that is :)

**Pull request closed without merging**